packages/core/src/utils.ts

```diff
@@ -0,0 +1,237 @@
+    } else {
+      throw new Error('Something went wrong while fetching env metadata');
+    }
+  }
+}
+
+function getIndexerURL(environment: Environment): string {
+  if (environment === Environment.TESTNET) {
+    return IndexerUrl.TESTNET;
+  } else if (environment === Environment.MAINNET) {
+    return IndexerUrl.MAINNET;
+  } else {
+    throw new Error('Invalid environment');
+  }
+}
+
+/**
+ * Returns all Sygma supported domains (networks).
+ * By default it will fetch domains configured for mainnet bridge deployment.
+ * Alternative option can be specified in options.env.
+ * You can filter domains with specifity types or ones that have at least one outbound route of specified type.
+ */
+export async function getDomains(options: {
+  routeTypes?: RouteType[];
+  environment: Environment;
+  networkTypes?: Network[];
+}): Promise<Domain[]> {
+  const config = new Config();
+  await config.init(options.environment);
+  const domains = config.getDomains(options);
+  return domains;
+}
+
+/**
+ * Returns  supported routes originating from given source domain.
+ * @param source Either caip2 identifier, chainId or sygmaId
+ * @param options Allows selecting bridge instance (mainnet by default) and filtering routes by type.
+ */
+export async function getRoutes(
+  source: Domainlike,
+  environment: Environment = process.env.SYGMA_ENV,
+  options?: {
+    routeTypes?: RouteType[];
+    sourceProvider?: Eip1193Provider;
+  },
+): Promise<Route[]> {
+  try {
+    const config = new Config();
+    await config.init(environment);
+    const domainConfig = config.findDomainConfig(source);
+    const indexerUrl = getIndexerURL(environment);
+    const typeQuery = options?.routeTypes ? `?resourceType=${options.routeTypes.join(',')}` : '';
+    const url = `${indexerUrl}/api/routes/from/${domainConfig.id}${typeQuery}`;
+    const response = await fetch(url);
+    const data = (await response.json()) as { routes: RouteIndexerType[] };
+
+    let routeFeeHandlerAddressesAndTypes: Array<
+      RouteIndexerType & { feeHandlerAddress: string } & { feeHandlerType: FeeHandlerType }
+    >;
+
+    if (domainConfig.type === Network.EVM && options?.sourceProvider) {
+      const routesWithHandlerAddresses = await getFeeHandlerAddressesOfRoutes({
+        routes: data.routes,
+        chainId: domainConfig.chainId,
+        bridgeAddress: domainConfig.bridge,
+        provider: options.sourceProvider,
+      });
+
+      routeFeeHandlerAddressesAndTypes = await getFeeHandlerTypeOfRoutes({
+        routes: routesWithHandlerAddresses,
+        chainId: domainConfig.chainId,
+        provider: options.sourceProvider,
+      });
+    }
+
+    return data.routes.map(route => {
+      const resource = domainConfig.resources.find(r => r.resourceId === route.resourceId)!;
+
+      let routeWithTypeAndAddress;
+      if (routeFeeHandlerAddressesAndTypes) {
+        routeWithTypeAndAddress = routeFeeHandlerAddressesAndTypes.find(r => {
```

Comment:
this is very golang way of writing a variable name. Can we rename `r` to something more semantic?

Reply:
open for suggestions

Reply:
I would just go with `route => `